src/librustc/ty/context.rs

```diff
@@ -836,11 +836,11 @@ pub struct GlobalCtxt<'tcx> {
     // Records the free variables refrenced by every closure
     // expression. Do not track deps for this, just recompute it from
     // scratch every time.
-    freevars: FxHashMap<HirId, Rc<Vec<hir::Freevar>>>,
+    freevars: FxHashMap<DefId, Rc<Vec<hir::Freevar>>>,
```

Comment:
I believe there's `DefIdMap` and `DefIdSet` aliases which could be used for these changes.